### PR TITLE
Support authenticating against docker hub

### PR DESCRIPTION
**Pull request by deviantintegral**

### CODE REVIEW

**Comment by achekulaev** [comment]
Nice change @deviantintegral. Thanks!

File: Vagrantfile [hunk 1]
```diff
@@ -229,6 +229,19 @@ Vagrant.configure("2") do |config|
     SCRIPT
   end
 
+  # Let users provide credentials to log in to Docker Hub.
+  if $vconfig['docker_registry_auth']
+
+    config.vm.provision "trigger", :option => "value" do |trigger|
+      trigger.fire do
+        info 'Authenticating with the Docker registry'
+        system "docker -H localhost:2375 login"
```
Comment (lmakarov) [inline Vagrantfile:238]: I'm not very clear with using triggers for provisioning.
Where will this run - host or VM?
I assume this need to run one time in the VM to pick up settings from config.json?

Comment (deviantintegral) [inline Vagrantfile:238]: This is run on the host, and then line 242 copies config.json (with the authentication tokens) into the VM.

**Comment by lmakarov** [comment]
Thanks @deviantintegral !
I had to manually merge this PR. Mentioned you in the commit message :)

**Pull request closed without merging**